### ISSUE TITLE
Unable to update theme after switching theme

### ISSUE THREAD

**Issue by billreed63** [maqetta/maqetta#3306]
1) Create a claro theme can change the button background image gradient
2)  Create a desktop html
3) DnD a button 
4) Expand the cascade on the button background image
5) Switch the theme to your custom claro theme
6) Not that the background image has the rule for the button image from your custom css selected but a duplicate new rule above it. 
7) Change the background image gradient  not that the button in the page is not update

If you refresh the page the button will reflect the change so it appears that the change is being made to the model and the css just the hot update of the CSS in the browser does not seem to be working 

Once the browser is refreshed change made to the button are made to the load CSS and well as the model and the duplicate new rule is no longer added.

I think this regression was introduced when fixing #3224 , where we removed the setText that forced the reload of the whole page.

**Comment by jhpedemonte** [comment]
@JonFerraiolo 

How big of an issue is this, for M7?  How risky would a fix be?

**Comment by JonFerraiolo** [comment]
I don't think very many people will hit this bug, and very high risk.

@billreed63 once you are done studying this, unless there are other factors that come up, feel free to push to M8

**Comment by JonFerraiolo** [comment]
Pushing to M8.

Investigated some more. Got an exception with these steps:
- Clone claro
- New Desktop App
- Add a Button
- Open up Background props
- Change theme to the clone
- Click on ... button next to background-image
- Choose linear-radient, click OK in background dialog

Stack trace:

```
declare.getCommandForStyleChange (ThemeModifier.js:156)
declare._stylePropertiesChange (VisualEditor.js:212)
declare._stylePropertiesChange (PageEditor.js:282)
lang.hitch (lang.js:375)
target.(anonymous function).dispatcher (aspect.js:94)
on.emit (on.js:274)
on.emit (on.js:324)
Evented.emit (Evented.js:31)
publish (topic.js:25)
connect.publish (connect.js:323)
declare._changeValue (Cascade.js:297)
innerChangeValueFunc (Cascade.js:184)
_submit (Cascade.js:222)
lang.hitch (lang.js:375)
DialogClass.showDialog (Dialog.js:257)
lang.hitch (lang.js:375)
target.(anonymous function).dispatcher (aspect.js:94)
declare._onSubmit (_DialogMixin.js:54)
declare._onClick (_ButtonMixin.js:48)
inherited (declare.js:189)
declare._onClick (Button.js:64)
lang.hitch (lang.js:373)
handles (a11yclick.js:84)
```

@billreed63 has investigated further, and the bug definitely is that setContent() isn't getting called to update the canvas and the model. Same problem for both original problem and the exception.

**Comment by billreed63** [comment]
related to #3167

**Comment by billreed63** [comment]
We do a setContent in ChangeThemeCommand the Editor setContent gets called... the textModel exists so we create another dojo.connection to the onChange of the model... when the model changes onChange is called when that fires onChanged (which if you follow it around does a setContent)   So everytime we do a setContent we add a onChange handler.. so the number of calls to onChange grows with each setContent... 

We do a setContent which triggers the old event handler that then does a setContent that triggers the 2nd onChanged whch does setContent (3rd onChange handle) triggering the first, which does a set triggering the 2 and 3 which do setContent and so one....
actually that is a simplified explanation.. because visualEditor is in the mix... the setContent that triggers the onChange comes from VE and when the setContent in Editor happens it triggers VE's onChange.... 

So by doing a dojo.disconnect on the old onChange handler in Editor.setContent and creating a new dojo.connection we ensure that we will only have one call to Editor.onChanged for each model change

**Comment by JonFerraiolo** [comment]
@billreed63 - Could you please review this issue to see if it is a candidate for M7.0.1? My guess is no, too risky, but you know this issue the best.

**Comment by billreed63** [comment]
Fixed by c28ec116ccd0c3ad87f17a694e4ec98eb04883e6
